docs/Documentation/Software_Tools/spack.md

```diff
@@ -0,0 +1,302 @@
+    paths:
+      slurm@18-08-0-3: /nopt/slurm/18.08.3
+      slurm@19-05-0-2: /nopt/slurm/19.05.2
+```
+
+that will enable builds against both installed Slurm versions. 
+Then you should see
+
+```bash
+[$user@el2 ~]$ spack spec openmpi@3.1.3%gcc@7.3.0 ^slurm@19-05-0-2
+Input spec
+--------------------------------
+openmpi@3.1.3%gcc@7.3.0
+    ^slurm@19-05-0-2
+
+Concretized
+--------------------------------
+openmpi@3.1.3%gcc@7.3.0 cflags="-O2 -march=skylake-avx512 -mtune=skylake-avx512" cxxflags="-O2 -march=skylake-avx512 -mtune=skylake-avx512" fflags="-O2 -march=skylake-avx512 -mtune=skylake-avx512" +cuda+cxx_exceptions fabrics=verbs ~java~legacylaunchers~memchecker+pmi schedulers=slurm ~sqlite3~thread_multiple+vt arch=linux-centos7-x86_64
+-
+    ^slurm@19-05-0-2%gcc@7.3.0 cflags="-O2 -march=skylake-avx512 -mtune=skylake-avx512" cxxflags="-O2 -march=skylake-avx512 -mtune=skylake-avx512" fflags="-O2 -march=skylake-avx512 -mtune=skylake-avx512" ~gtk~hdf5~hwloc~mariadb+readline arch=linux-centos7-x86_64
+```
+
+where the Slurm dependency will be satisfied with the installed Slurm (cflags, cxxflags, and arch are coming from site-wide configuration in `/nopt/nrel/apps/base/2018-12-02/spack/etc/spack/compilers.yaml`; the variants string is likely coming from the configuration in the Spack database, and should be ignored).
+
+## Virtual Packages 
+
+It is possible to specify some packages for which multiple options are available at a higher level. 
+For example, `mpi` is a virtual package specifier that can resolve to mpich, openmpi, Intel MPI, etc. 
+If a package's dependencies are spec'd in terms of a virtual package, Spack will choose a specific package at build time according to site preferences.
+Choices can be constrained by spec, e.g.,
+
+`spack install X ^mpich@3`
+
+would satisfy package X’s mpi dependency with some version 3 of MPICH.
+You can see available providers of a virtual package with
+
+`spack providers <vpackage>`
+
+## Extensions
+
+In many cases, frameworks have sub-package installations in standard locations within their own installations. 
+A familiar example of this is Python and its usual module location in `lib(64)/python<version>/site-packages`, and pointed to via the environment variable `PYTHONPATH`.
+
+To find available extensions
+
+`spack extensions <package>`
+
+Extensions are just packages, but they are not enabled for use out of the box. To do so (e.g., so that you could load the Python module after installing), you can either load the extension package’s environment module, or
+
+`spack use <extension package>`
+
+This only lasts for the current session, and is not of general interest. A more persistent option is to activate the extension:
+
+`spack activate <extension package>`
+
+This takes care of dependencies as well. The inverse operation is deactivation.
+
+
+|command | description |
+|-|-|
+|`spack deactivate <extension package>`	        |extension alone. Will not deactivate if dependents exist |
```

Comment:
could be changed to "deactivates extension alone" to match formatting of other rows.